### PR TITLE
US6: Added delete function and alert pop up

### PR DESCRIPTION
**Pull request by KevinKyaterekera**
I just corrected the form validation with patterns, an altert and end date cannot be before start date anymore

### CODE REVIEW

File: components/Form.js [hunk 1]
```diff
@@ -9,10 +9,15 @@ export default function Form({ currentTrips, setCurrentTrips }) {
     const newTripObject = {
       tripName: data.name,
```
Comment (alperkuertuel) [inline components/Form.js:13]: Maybe you can give your `newTripObject` an id key to later use it as a valid unique key. Can it happen, that some trips are called the same? But are just in another time span with another tripUser?

File: components/TravelDetails/index.js [hunk 2]
```diff
@@ -1,23 +1,27 @@
...
-        📍{trip.destination}
-      </article>
-      <FlexArticle>
-        <UserIcon />
-        <TimePeriod />
-      </FlexArticle>
-    </FlexContainer>
+    <>
+      {currentTrips.map((trip) => (
+        <FlexContainer key={trip.name}>
```
Comment (alperkuertuel) [inline components/TravelDetails/index.js]: If some trips would possibly have the same name, your key won't be unique. But you can also validate your trip name in the form that a trip name has already been given.

File: components/Form.js [hunk 3]
```diff
@@ -37,7 +42,7 @@ export default function Form({ currentTrips, setCurrentTrips }) {
         Start date:
         <StyledInput
           type="date"
-          name="date"
+          name="startdate"
           placeholder="Choose a start date"
```
Comment (alperkuertuel) [inline components/Form.js:59]: I had a tremendous stress going through the QA without a type of restriction or validation from user inputs fields. Is this planned in future?

**Review by alperkuertuel — COMMENTED** [review]
Everything looks fine and should work as coded! I just left you some comments and Ideas, which can fasten up going through the QA.

File: components/Form.js [hunk 4]
```diff
@@ -1,69 +1,92 @@
 import { styled } from "styled-components";
 import PrimaryLink from "./PrimaryLink";
+import { uid } from "uid";
+const slugify = require("slugify");
```
Comment (F-Kirchhoff) [inline components/Form.js]: please use the standart way to import libraries with `import <name> from "package-name"`

File: components/Form.js [hunk 5]
```diff
@@ -1,69 +1,92 @@
...
+      slug: slugify(data.name, { lower: true }),
       tripDestination: data.destination,
+      tripUser: data.user,
+      tripStart: data.startdate,
+      tripEnd: data.enddate,
     };
-    setCurrentTrips([...currentTrips, newTripObject]);
-    event.target.reset();
+    if (data.enddate < data.startdate) {
+      window.alert("End date ca not be before start date!");
```
Comment (F-Kirchhoff) [inline components/Form.js]: typo

File: components/TripDetailView.js [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  /* within this function we need a dialog that asks if we really want to delete this trip*/
+  const handleOpenDialog = () => {
+    deleteDialog.showModal();
+  };
+  return (
+    <>
+      <FlexContainer key={trip.tripId}>
+        <article>
+          {trip.tripName}
+          <br />
```
Comment (F-Kirchhoff) [inline components/TripDetailView.js]: please don't use br tags

**Review by F-Kirchhoff — APPROVED** [review]
LGTM, there is only one issue with your import of 'slugify'. Please change it and then you can go directly to QA.

**Pull request merged**